### PR TITLE
Remove unnecessary CDTs

### PR DESCRIPTION
**Pull request by hmaarrfk**
rebased onto main

* Closes https://github.com/conda-forge/qt-main-feedstock/pull/114
* Closes #122 

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by hmaarrfk** [comment]
@isuruf any clue which library is missing in the test section? I tried to keep both sections having the same CDTs.

```
undefined reference to `_XReply'
```

File: recipe/meta.yaml [hunk 1]
```diff
@@ -185,35 +157,12 @@ requirements:
...
-    - {{ cdt('libxdamage') }}            # [linux]
-    - {{ cdt('libxdamage-devel') }}      # [linux]
-    - {{ cdt('libxfixes') }}             # [linux]
-    - {{ cdt('libxfixes-devel') }}       # [linux]
-    - {{ cdt('libxxf86vm') }}            # [linux]
-    - {{ cdt('libxcb') }}                # [linux]
-    - {{ cdt('expat-devel') }}           # [linux]
-    - {{ cdt('pcre') }}                  # [linux]
     - {{ cdt('libglvnd-glx') }}          # [linux]
+    - {{ cdt('libglvnd-egl') }}          # [linux]
```
Comment (isuruf) [inline recipe/meta.yaml:165]: You need `cdt('libx11')`. There are some wrong dependencies in `mesa-libgl`

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

File: recipe/meta.yaml (hunk 1 above)
Comment (hmaarrfk) [reply recipe/meta.yaml:165]: Did you already check if any of these CDTs can be removed? If so, I'll just give up on this extra cleaning

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

File: recipe/meta.yaml (hunk 1 above)
Comment (hmaarrfk) [reply recipe/meta.yaml:179]: I think i found a solution, but it requires one new xorg package.
https://github.com/conda-forge/staged-recipes/pull/22084

File: recipe/yum_requirements.txt [hunk 2]
```diff
@@ -1,23 +1,6 @@
...
 mesa-libGL
+mesa-libEGL-devel
 mesa-dri-drivers
-libselinux
-libXdamage
-libXxf86vm
-libXext
-libX11-devel
-libXt-devel
 libXext-devel
```
Comment (hmaarrfk) [inline recipe/yum_requirements.txt:4]:
```suggestion
```

**Comment by hmaarrfk** [comment]
This is finally ready for review.

The main challenge is that EGL and GL require some X11 libraries so that `pkg-config --exists egl` passes. I found these libraries by dropping into the debug shell using `python build-locally.py` and using

```
pkg-config --debug --exists egl
```

and installed conda(-forge) packages as necessary.

I also added flags to the configure script to ensure that future maintainers cannot easily remove X11 dependencies without also breaking EGL

File: recipe/meta.yaml [hunk 3]
```diff
@@ -164,6 +126,23 @@ requirements:
     - alsa-lib                           # [linux]
     - krb5
     - harfbuzz                           # [linux]
+    # Adding the following xcb-util run requirements resolves issue
+    # that may exhibit as segfault due to missing libraries
+    # in some lighter weight linux installations
+    # https://github.com/conda-forge/qt-main-feedstock/issues/32
```
Comment (ccordoba12) [inline recipe/meta.yaml:132]:
```suggestion
    # The xcb-util packages are necessary to build the Qt xcb plugin.
```

I think a comment like this is a better suited for the `host` section (you can reedit it if you want).

File: recipe/run_test.sh [hunk 4]
```diff
@@ -4,19 +4,24 @@ set -ex
...
 
-ls
 cd test
-ln -s ${GXX} g++
+ln -sf ${GXX} g++
 cp ../xcrun .
 cp ../xcodebuild .
 export PATH=${PWD}:${PATH}
-qmake hello.pro
+# To learn about qmake flags, read
```
Comment (ccordoba12) [inline recipe/run_test.sh:12]:
```suggestion

# To learn about qmake flags, read
```

**Review by ccordoba12 — COMMENTED** [review]
Some small suggestions for you @hmaarrfk, otherwise looks good to me.

File: recipe/build.sh [hunk 5]
```diff
@@ -120,9 +120,12 @@ if [[ $(uname) == "Linux" ]]; then
                 -system-zlib \
                 -system-harfbuzz \
                 -system-sqlite \
+                -system-harfbuzz \
```
[the remark this answers is not in the document]
Comment (hmaarrfk) [reply recipe/build.sh:123]: I was really blind

File: recipe/meta.yaml (hunk 3 above)
Comment (hmaarrfk) [reply recipe/meta.yaml:132]: done in the next push.

File: recipe/run_test.sh (hunk 4 above)
Comment (hmaarrfk) [reply recipe/run_test.sh:12]: new line? done.

**Comment by hmaarrfk** [comment]
Given that this move is likely contentious, but also a very valuable cleanup, I would like to move forward on it on March 12 (2 weeks from now). It should give enough time for all to provide feedback on the current proposal.

**Comment by eddy-geek** [comment]
If I understand correctly, the build failures are the usual timeouts, so nothing is blocking this from being merged on March 12th  as said above (https://github.com/conda-forge/qt-main-feedstock/pull/115#issuecomment-1445171643) ?
(I'm just tracking it as a gdal user as it appears to be a dependency for poppler and in turn conda-forge/gdal-feedstock#486 -- if I read the [status](https://conda-forge.org/status/#other_migrations) correctly)

**Comment by hmaarrfk** [comment]
> March 12th

correct. i am also building the other architectures right now (with emulation) so i'm really hoping that things are ready to go at this stage.

I do want to give "a saturday" for any final comments, but I think we really worked through many of the issues to make this a "clear improvement".

**Comment by hmaarrfk** [comment]
builds at https://anaconda.org/mark.harfouche/qt-main/files
logs  [qt-log.zip](https://github.com/conda-forge/qt-main-feedstock/files/10946464/qt-log.zip)

**Pull request merged**